lib/bank_account/savings_account.rb

```diff
@@ -0,0 +1,28 @@
+module Bank
+  class SavingsAccount < Account
+
+    def initialize(id, initial_balance)
+      super
+      raise ArgumentError.new "Insufficient Funds" unless initial_balance >= 10
+    end
+
+    # charges $2 for every withdrawal
+    def withdraw(amount)
+      @balance -= (amount + 2)
+      if @balance < 10
+        puts "Savings can not fall below $10.Your remaining savings amount is: "
+        return @balance += (amount + 2)
+      else
+        return @balance
+      end
+    end
+
+    # calculates interest on balance where rate is in %
+    def add_interest(rate)
+      @accrued_interest = @balance * (rate/100.to_f)
+      @balance += @accrued_interest
+      return @accrued_interest
+    end
```

Comment:
I'm not convinced you need to make `accrued_interest` an instance variable (the `@` turns it into one), since you aren't using it outside of this method and you reset it every time this method is called. Also Ruby should handle the conversion to a float if it's needed on line 22, so you don't really need to do the extra conversion at the end there.